src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -387,16 +363,16 @@ public int compare(TypedVar o1, TypedVar o2) {
 
   // Due to lack of precise definition of a namespace, we look for object types that are not of any
   // other type. This will need more refinement as more cases appear.
-  private boolean isLikelyNamespace(JSType type, Compiler compiler) {
+  private boolean isLikelyNamespace(JSType type) {
     UnionType utype = type.toMaybeUnionType();
     if (utype != null) {
-      return isNativeObjectType(utype.restrictByNotNullOrUndefined(), compiler);
+      return isNativeObjectType(utype.restrictByNotNullOrUndefined());
     }
-    return isNativeObjectType(type, compiler);
+    return isNativeObjectType(type);
   }
 
   // TODO(rado): refactor so that the native object type is a final field.
-  private boolean isNativeObjectType(JSType type, Compiler compiler) {
```

Comment:
Great, this was really bothering me. Much cleaner now, to have the lifetime of Generator match the lifetime of the compiler.